doc/releases/release-notes-4.0.rst

```diff
@@ -319,6 +424,7 @@ Drivers and Sensors
   * Fixed calibration scheme in ESP32-S3.
   * STM32H7: Added support for higher sampling frequencies thanks to boost mode implementation.
   * Added initial support for Renesas RA8 ADC driver (:dtcompatible:`renesas,ra-adc`)
+  * Added support for NXP S32 SAR_ADC
```

Comment:
```suggestion
  * Added support for NXP S32 SAR_ADC (:dtcompatible:`nxp,s32-adc-sar`).
```

in general, it would be nice to add references to the compatible whenever mentioning additions/changes to drivers in your PR